src/resonate/scheduler/scheduler.py

```diff
@@ -115,41 +126,103 @@ def run(
         **kwargs: P.kwargs,
     ) -> Handle[T]:
         # If there's already a record with this ID, dedup.
-        record = self._records.get(id)
-        if record is not None:
-            return record.handle
+        handle = Handle[T](id)
+        self._cmd_queue.put(Onboard(id, handle, Invocation(func, *args, **kwargs)))
+        return handle
+
+    def _heartbeat(self) -> None:
+        assert isinstance(self._store, RemoteStore)
+        while True:
+            affected: int | None = None
+            with contextlib.suppress(requests.exceptions.ConnectionError):
+                affected = self._store.tasks.heartbeat(pid=self._pid)
+                logger.debug("Heatbeat affected %s tasks", affected)
+            time.sleep(2)
+
+    def _loop(self) -> None:
+        while True:
+            cmd = self._cmd_queue.get()
+            if cmd is None:
+                break
+
+            # start the next tick
+            for loopback in self._step(cmd):
+                self._cmd_queue.put(loopback)
+
+            # mark task done
+            self._cmd_queue.task_done()
+
+    def _step(self, cmd: Command) -> list[Command]:  # noqa: PLR0911
+        if isinstance(cmd, Invoke):
+            return self._handle_invoke(cmd)
+        if isinstance(cmd, Resume):
+            return self._handle_resume(cmd)
+        if isinstance(cmd, Complete):
+            return self._handle_complete(cmd)
+        if isinstance(cmd, Claim):
+            return self._handle_claim(cmd)
+        if isinstance(cmd, Onboard):
+            return self._handle_onboard(cmd)
+        if isinstance(cmd, Suscribe):
+            return self._handle_suscribe(cmd)
+        if isinstance(cmd, Notify):
+            return self._handle_notify(cmd)
+        assert_never(cmd)
+
+    def _handle_notify(self, notify: Notify) -> list[Command]:
```

Comment:
This is incredibly pedantic but would you mind defining these functions in the same order as the if statements in the _step function?